cmake/ConfigureCompiler.cmake

```diff
@@ -11,6 +11,13 @@ set(USE_LIBCXX OFF CACHE BOOL "Use libc++")
 set(USE_CCACHE OFF CACHE BOOL "Use ccache for compilation if available")
 set(RELATIVE_DEBUG_PATHS OFF CACHE BOOL "Use relative file paths in debug info")
 set(STATIC_LINK_LIBCXX ON CACHE BOOL "Statically link libstdcpp/libc++")
+set(USE_WERROR OFF CACHE BOOL "Compile with -Werror. Recommended for local development and CI.")
+
+# Default -Werror for environments where we're committed to compiling warning-free
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION STREQUAL "8.3.1")
+  message(STATUS "We officially support -Werror for ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION} - USE_WERROR is ON")
+  set(USE_WERROR ON)
+endif()
```

Comment:
I think rather than this, I'd prefer you to add -DUSE_WERROR=ON to build/docker-compose.yaml so that it's enabled by default for builds and CI runs, and add it to the README as a thing that developers should probably enable in their local cmake invocations, the same way that we handle USE_CCACHE now.

Reply:
Sorry, my rational here is that this is going to go out of date the instant any GCC version is touched, e.g. the current 3 toolchain state we're in, at which point we're in no better of a state than what I've described, except that it wouldn't be in the docs/readme anyway.

Reply:
Done